OpenSSL/crypto.py

```diff
@@ -172,15 +176,18 @@ def generate_key(self, type, bits):
         """
         Generate a key of a given type, with a given number of a bits
 
-        :param type: The key type (TYPE_RSA or TYPE_DSA)
-        :param bits: The number of bits
+        :param type: The key type (TYPE_RSA or TYPE_DSA or TYPE_EC)
+        :param bits: The number of bits or EllipticCurve
```

Comment:
I agree, it's not the greatest approach.

The current implementation is used as follows:
```
curve = OpenSSL.crypto.get_elliptic_curve('secp384r1')
pkey = PKey()
pkey.generate(key, TYPE_EC, curve)
```

Honestly I think the best approach would be to subclass `PKey` into  `RSAPKey`, `DSAPKey`, and `ECPKey` but that is getting away from the thin abstraction on top of OpenSSL.